PWGHF/TableProducer/candidateCreatorDstar.cxx

```diff
@@ -173,9 +175,18 @@ struct HfCandidateCreatorDstar {
       auto bc = collision.template bc_as<aod::BCsWithTimestamps>();
       if (runNumber != bc.runNumber()) {
         // LOG(info) << ">>>>>>>>>>>> Current run number: " << runNumber;
-        initCCDB(bc, runNumber, ccdb, isRun2 ? ccdbPathGrp : ccdbPathGrpMag, lut, isRun2); // Sets up the grp object for magnetic field (w/o matCorr for propagation)
+        o2::parameters::GRPMagField* grpo = ccdb->getForTimeStamp<o2::parameters::GRPMagField>(ccdbPathGrpMag, bc.timestamp());
+        if (grpo == nullptr) {
+          LOGF(fatal, "Run 3 GRP object (type o2::parameters::GRPMagField) is not available in CCDB for run=%d at timestamp=%llu", bc.runNumber(), bc.timestamp());
+        }
+        o2::base::Propagator::initFieldFromGRP(grpo);
         bz = o2::base::Propagator::Instance()->getNominalBz();
         // LOG(info) << ">>>>>>>>>>>> Magnetic field: " << bz;
+        runNumber = bc.runNumber();
+
+        // initCCDB(bc, runNumber, ccdb, isRun2 ? ccdbPathGrp : ccdbPathGrpMag, lut, isRun2); // Sets up the grp object for magnetic field (w/o matCorr for propagation)
+        // bz = o2::base::Propagator::Instance()->getNominalBz();
+        // // LOG(info) << ">>>>>>>>>>>> Magnetic field: " << bz;
```

Comment:
Please use `initCCDB`. The `lut` pointer is optional.

Reply:
Any change related to ```lut``` came from my previous closed (PR)[https://github.com/AliceO2Group/O2Physics/pull/4866]. Now I merged master branch and these will not appear on changed files. 
Only changes in candidateCreatorDstar.cxx are done in lines # 321, 105.

Reply:
So you will fix it in another PR?

Reply:
It would be good to separate the changes useful for correlation study and MatCorr( or lut). As soon as this PR is merged, I can create another PR for this.

Reply:
OK, thanks

Reply:
thanks